File: valid_anagram.go
package main

import "fmt"

func is_pair_anagram(str1, str2 string) {
	map1 := make(map[string]int)
	map2 := make(map[string]int)

	if len(str1) != len(str2) {
		fmt.Println(false)
		return
	}
	for id, val := range str1 {
		_, ok := map1[string(val)]
		if ok {
			map1[string(val)] += 1
		} else {
			map1[string(val)] = 1
		}
		_, ok = map2[string(str2[id])]
		if ok {
			map2[string(str2[id])] += 1
		} else {
			map2[string(str2[id])] = 1
		}
	}
	for key, _ := range map1 {
		_, ok := map2[key]
		if ok {
			if map1[key] != map2[key] {
				fmt.Println(false)
				return
			}
		} else {
			fmt.Println(false)
			return
		}
	}
	fmt.Println(true)
	return
}

func main() {
	is_pair_anagram("hard work pay's off", "pay's off hard work")
}

File: valid_anagram.py
from collections import defaultdict


map1 = defaultdict(int)
map2 = defaultdict(int)


def is_pair_anagram(str1,str2):
    if len(str1) != len(str2):
        return False
    
    for id,chr in enumerate(str1):
        map1[chr] += 1
        map2[str2[id]] += 1

    for element in map1:
        if map1[element] != map2[element]:
            return False
    return True



def main():
    result = is_pair_anagram('bav credit','debit card')
    print(result)


main()